lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/service/LessonBuilderEntityProducer.java

```diff
@@ -1006,31 +1033,41 @@ else if (type == SimplePageItem.ASSESSMENT)
        return needFix;
     }
 
-    String mergeGroups(Element itemElement, String attr, Collection<Group> siteGroups) {
+    String mergeGroups(Element itemElement, String attr, Collection<Group> siteGroups, String fromSiteId) {
 
 	// not currently doing this, although the code has been tested.
 	// The problem is that other tools don't do it. Since much of our group
 	// awareness comes from the other tools, enabling this produces
 	// inconsistent results
 
-	NodeList groups = itemElement.getElementsByTagName(attr);
+	String groups = itemElement.getAttribute(attr);
 	String groupString = null;
-	
-	// translate groups from title to ID
-	if (groups != null && siteGroups != null) {
-	    for (int n = 0; n < groups.getLength(); n ++) {
-		Element group = (Element)groups.item(n);
-		String title = group.getAttribute("title");
-		if (title != null && !title.equals("")) {
-		    for (Group g: siteGroups) {
-			if (title.equals(g.getTitle())) {
-			    if (groupString == null)
-				groupString = g.getId();
-			    else
-				groupString = groupString + "," + g.getId();
+	Site oldSite = null;
+	try {
+		oldSite = siteService.getSite(fromSiteId);
+	} catch (Exception e) {
+		log.debug("site " + fromSiteId + " not found.", e);
+	}
+	// For each old group, get the corresponding new group id
+	if (!groups.isEmpty() && siteGroups != null && oldSite != null) {
+		final String[] parts = groups.split(",");
+		for (int n = 0; n < parts.length; n ++) {
```

Comment:
I'd just use for (String part : parts) { here. Clearer, unless you need to test the index.